lib/portlayer/exec/commit_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	
+	"github.com/vmware/vic/pkg/trace"
+)
+
+func TestBatchBlockOnFunc(t *testing.T) {
```

Comment:
It would be good to also include a test of the error handling: if an operation produces an error, does that get properly passed to the caller?